bash_completion

```diff
@@ -1209,108 +1209,102 @@ __parse_options()
     if [[ $option =~ (\[((no|dont)-?)\]). ]]; then
         option2=${option/"${BASH_REMATCH[1]}"/}
         option2=${option2%%[<{().[]*}
-        printf '%s\n' "${option2/=*/=}"
+        _options+=("${option2/=*/=}")
         option=${option/"${BASH_REMATCH[1]}"/"${BASH_REMATCH[2]}"}
     fi
 
     option=${option%%[<{().[]*}
     option=${option/=*/=}
     [[ $option ]] || return 1
 
-    printf '%s\n' "$option"
+    _options+=("$option")
 }
 
-# Parse GNU style help output of the given command.
-# @param $1  command; if "-", read from stdin and ignore rest of args
-# @param $2  command options (default: --help)
-#
-# TODO: rename per API conventions, rework to use vars rather than outputting
-_parse_help()
+# Parse GNU style help output of the given command and generate and store
+# completions in an array.
+# Usage:
+#   _comp_compgen_help [-u] command [opt]
+# Options:
+#   -u        Parse the BSD style usage output.  The default is in the GNU
+#             style help output.
+# @param $1 command  command; if "-", read from stdin and ignore rest of args
+# @param $2 opt      command options (default: --help)
+_comp_compgen_help()
```

Comment:
Would this be something intended to be called through the `_comp_compgen help` split behavior? If yes, any particular reason we are not calling it like that in the changes here, but rather as `_comp_compgen_help`? The calling convention for the filedir one throughout the tree is to use the split form.

Aside, are `_comp_compgen_set` and `_comp_compgen_tilde` intended to be callable using the split form? They're being called using the non-split form at the moment. Wondering if some of these functions are inadvertently callable using `_comp_compgen NAME`.

Reply:
I originally thought to allow both forms when there are no additional options (`_comp_compgen -aRvc`) so that  `_comp_compgen_filedir`/`_comp_compgen filedir`, `_comp_compgen_help`/`_comp_compgen help`, etc.,  are equivalent. `_comp_compgen_set` and `_comp_compgen_tilde` are also intended to be used with `_comp_compgen` when there are additional options to `_comp_compgen`.

The reason that I used the non-split form was that it is faster.  On the other hand,  `_filedir` was once converted to `_comp_compgen -a filedir`, and then `-a` was removed afterward, so it ended up in `_comp_compgen filedir`.

I think it is good to normalize it. It seems the split form, like `_comp_compgen help`, `_comp_compgen set`, and `_comp_compgen tilde` is better for consistency. But maybe the non-split form (when there are no options, `-aRvc`) could also be an option for efficiency. I can normalize it in either direction. What is your preference?

Reply:
I think it's easy enough to remember that if no `_comp_compgen` features are needed, calling `_comp_compgen_FOO` directly is ok (and preferred) for efficiency. So I guess that's what I'd prefer, but whatever we go with would be good to be documented somewhere, e.g. in the style guide.

Reply:
The change `_comp_compgen filedir` => `_comp_compgen_filedir` is another large one and makes the GitHub interface unuseful, so I decided to separate the PR for it.

Reply:
Created in #964